File: {{ cookiecutter.app_name }}/{{ cookiecutter.lib_name }}/src/main/java/{{ cookiecutter.lib_package_dir }}/Callback.java
package {{ cookiecutter.lib_package_name }};

/**
 *
 * @author {{ cookiecutter.author_name }}
 * @version {{ cookiecutter.version_package_name }} {% now 'local', '%d/%m/%Y' %}
 * @since {{ cookiecutter.version_package_name }}
 */
public abstract class Callback<T extends DataResponse> {

    public abstract void throwError(APIException exception);

    public abstract void response(T bean);
}

File: {{ cookiecutter.app_name }}/{{ cookiecutter.lib_name }}/src/main/java/{{ cookiecutter.lib_package_dir }}/Callback.kt
package {{ cookiecutter.lib_package_name }};

/**
 *
 * @author {{ cookiecutter.author_name }}
 * @version {{ cookiecutter.version_package_name }} {% now 'local', '%d/%m/%Y' %}
 * @since {{ cookiecutter.version_package_name }}
 */
abstract class Callback<T : DataResponse?> {
    abstract fun throwError(exception: APIException?)
    abstract fun response(bean: T)
}
